packages/agreement-process/src/services/agreementContractBuilder.ts

```diff
@@ -353,12 +308,57 @@ const getPdfPayload = async (
     eServiceName: eservice.name,
     producerText,
     consumerText,
-    certifiedAttributes: getCertifiedAttributeHtml(certified),
-    declaredAttributes: getDeclaredAttributeHtml(declared),
-    verifiedAttributes: getVerifiedAttributeHtml(verified),
+    certifiedAttributes: certified.map(({ attribute, tenantAttribute }) => ({
+      assignmentDate: dateAtRomeZone(tenantAttribute.assignmentTimestamp),
+      assignmentTime: timeAtRomeZone(tenantAttribute.assignmentTimestamp),
+      attributeName: attribute.name,
+      attributeId: attribute.id,
+    })),
+    // eslint-disable-next-line sonarjs/no-identical-functions
+    declaredAttributes: declared.map(({ attribute, tenantAttribute }) => ({
+      assignmentDate: dateAtRomeZone(tenantAttribute.assignmentTimestamp),
+      assignmentTime: timeAtRomeZone(tenantAttribute.assignmentTimestamp),
+      attributeName: attribute.name,
+      attributeId: attribute.id,
+    })),
+    verifiedAttributes: verified.map(({ attribute, tenantAttribute }) => {
+      const expirationDate = getVerifiedAttributeExpirationDate(
+        tenantAttribute,
+        producer.id
+      );
+      return {
+        assignmentDate: dateAtRomeZone(tenantAttribute.assignmentTimestamp),
+        assignmentTime: timeAtRomeZone(tenantAttribute.assignmentTimestamp),
+        attributeName: attribute.name,
+        attributeId: attribute.id,
+        expirationDate: expirationDate
+          ? dateAtRomeZone(expirationDate)
+          : undefined,
+      };
+    }),
   };
 };
 
+function getVerifiedAttributeExpirationDate(
+  tenantAttribute: VerifiedTenantAttribute,
+  producerId: TenantId
+): Date | undefined {
+  const activeProducerVerification = tenantAttribute.verifiedBy
+    .filter((verification) => verification.id === producerId)
+    .sort((a, b) => a.verificationDate.getTime() - b.verificationDate.getTime())
+    .find(
+      (verification) =>
+        !tenantAttribute.revokedBy.find(
```

Comment:
```suggestion
        !tenantAttribute.revokedBy.some(
```
Since we don't need the return here we can use `some`